### PR TITLE
[ios] Fix external sku token getter

### PR DESCRIPTION
**Pull request by frederoni**
~`performSelector` doesn't support return values.~ `methodForSelector` doesn't seem to return any values. We can use NSInvocation’s `getReturnValue`, Swift’s `Selector.takeUnretainedValue()`, `performSelector`, or `-[NSObject valueForKeyPath:]`, which is the approach I went with here.

cc @friedbunny

### CODE REVIEW

**Comment by frederoni** [comment]
ready for review

cc @julianrex @friedbunny

File: platform/darwin/src/MGLAccountManager.m [hunk 1]
```diff
@@ -109,13 +96,14 @@ + (NSURL *)apiBaseURL {
...
 + (NSString *)skuToken {
-    if (MGLAccountManager.sharedManager.useExternalAccountManager) {
-        id externalAccountsClass = NSClassFromString(MGLAccountManagerExternalClassName);
-        SEL externalSKUTokenMethod = NSSelectorFromString(MGLAccountManagerExternalMethodName);
-        return ((NSString *(*)(id, SEL))[externalAccountsClass methodForSelector:externalSKUTokenMethod])(externalAccountsClass, externalSKUTokenMethod);
-    } else {
-        return MBXSKUToken.skuToken;
+    Class mbx = NSClassFromString(MGLAccountManagerExternalClassName);
+    if (mbx) {
+        if ([mbx respondsToSelector:NSSelectorFromString(MGLAccountManagerExternalMethodName)]) {
```
Comment (julianrex) [inline platform/darwin/src/MGLAccountManager.m]: The single `if (mbx) {` above is redundant.

**Comment by frederoni** [comment]
> I thought it did, as long as the return value is an object?

oh, you're right. It's just the delayed or threading `performSelector` methods that don't support return values. Would you prefer `performSelector` over `valueForKeyPath:`?

**Comment by julianrex** [comment]
No preference - I don't think there's a big diff in this context.

**Pull request merged**